### PR TITLE
[DX-1097] Update SessionState protobuf documentation to include BasicAuthData and example hash algorithms

### PR DESCRIPTION
**Pull request by dcs3spp**
### For internal users - Please add a Jira DX PR ticket to the subject!
[DX-1097](https://tyktech.atlassian.net/browse/DX-1097)

---

### Preview Link

[preview](https://deploy-preview-4324--tyk-docs.netlify.app/docs/nightly/plugins/supported-languages/rich-plugins/rich-plugins-data-structures/#sessionstate-session_stateproto)

### Description

Update SessionState protobuf documentation to include BasicAuthData and example hash algorithms

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR

[DX-1097]: https://tyktech.atlassian.net/browse/DX-1097?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

___

## **Type**
documentation

___

## **Description**
- Added detailed documentation for `BasicAuthData` within the SessionState protobuf, including fields and example hashing algorithms.
- Clarified that `basic_auth_data` contains a hashed password and the hashing method, enhancing the understanding of basic auth data handling.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>rich-plugins-data-structures.md</strong><dd><code>Enhanced SessionState Protobuf Documentation with BasicAuthData </code><br><code>Details</code></dd></summary>
<hr>

tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md
<li>Added a detailed description for <code>BasicAuthData</code> in the SessionState <br>protobuf documentation.<br> <li> Updated the <code>basic_auth_data</code> section to clarify it contains a hashed <br>password and the hashing method used.<br> <li> Included examples of hashing algorithms.

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4324/files#diff-145c1da09aa93cd7b1fbd3d9c15f65dbc8f633eaac8461743fbe2fe98dd81037">+22/-2</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md [hunk 1]
```diff
@@ -359,3 +360,23 @@ A list of headers, each header in this list is a structure that consists of two
 The key is a string that denotes the name of the header, the values are a list of strings that hold the content of the header, this is useful when the header has multiple associated values.
 This field is available for Go, Python and Ruby since tyk v5.0.4 and  5.1.1+.
 
+
+### BasicAuthData (coprocess_session_state.proto)
+
+The `BasicAuthData` contains a hashed password and the name of the hashing algorithm used. This is represented by the `basic_auth_data` attribute in [SessionState](#sessionstate-session_stateproto) message.
```
Comment (letzya) [inline tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md]:
```suggestion
The `BasicAuthData` contains a hashed password and the name of the hashing algorithm used. This is represented by the `basic_auth_data` attribute in [SessionState](#session-state) message.
```

File: tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md [hunk 2]
```diff
@@ -359,3 +360,23 @@ A list of headers, each header in this list is a structure that consists of two
...
+
+```yaml
+"basicAuthData": {
+    "password": "a_hashed_password_presentation",
```
Comment (letzya) [inline tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md]:
```suggestion
    "password": <a_hashed_password_presentation>,
```

Comment (letzya) [inline tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md]: I think that it's safer to use something that is not valid yaml so users will remember to change it. We use this format throughout the docs.

File: tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md [hunk 3]
```diff
@@ -359,3 +360,23 @@ A list of headers, each header in this list is a structure that consists of two
...
+    "hash": "the_hashing_algorithm_used_to_hash_the_password"
```
Comment (letzya) [inline tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md]:
```suggestion
    "hash": <the_hashing_algorithm_used_to_hash_the_password>
```

**Review by letzya — APPROVED** [review]

File: tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md [hunk 4]
```diff
@@ -359,3 +360,23 @@ A list of headers, each header in this list is a structure that consists of two
 The key is a string that denotes the name of the header, the values are a list of strings that hold the content of the header, this is useful when the header has multiple associated values.
 This field is available for Go, Python and Ruby since tyk v5.0.4 and  5.1.1+.
 
+
+### BasicAuthData (coprocess_session_state.proto)
+
+The `BasicAuthData` contains a hashed password and the name of the hashing algorithm used. This is represented by the `basic_auth_data` attribute in [SessionState](#session-state) message.
+
+```yaml
```
Comment (titpetric) [inline tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md:355]:
```suggestion
```json
```

File: tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md [hunk 5]
```diff
@@ -359,3 +360,23 @@ A list of headers, each header in this list is a structure that consists of two
...
+}
+```
+
+#### Field Descriptions
+
+`password`
+A hashed password.
+
+`hash`
+Name of the hashing algorithm used to hash the password, e.g. *bcrypt*, *Argon2*.
```
Comment (titpetric) [inline tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md]:
```suggestion
Name of the hashing algorithm used to hash the password.
```

First time ever hearing of argon2, unless we list explicit values, we shouldn't make up examples.

**Review by titpetric — APPROVED** [review]

File: tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md (hunk 5 above)
Comment (dcs3spp) [reply tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md]: Thanks @titpetric 

I have looked through the source on GHub and found [this](https://github.com/TykTechnologies/tyk/blob/ebe349cbf16e31a96067486291a91893cf5c7a0a/config/config.go#L679)

Is the suggestion below ok?

```suggestion
Name of the hashing algorithm used to hash the password, e.g. *murmur64*, *murmur128*, *sha256* or *bcrypt*
```

Comment (titpetric) [reply tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md]: I'd prefer you omit the details, and if necessary link https://tyk.io/docs/basic-config-and-security/security/key-hashing/ for more information.

Comment (dcs3spp) [reply tyk-docs/content/plugins/supported-languages/rich-plugins/rich-plugins-data-structures.md]: > I'd prefer you omit the details, and if necessary link https://tyk.io/docs/basic-config-and-security/security/key-hashing/ for more information.

Thanks @titpetric I've updated it in the branch as detailed below and will merge and release:

_Name of the [hashing algorithm]({{< ref "basic-config-and-security/security/key-hashing" >}}) used to hash the password._

**Pull request merged**